tests/structured_tests/split_10.toml

```diff
@@ -0,0 +1,21 @@
+query = """
+* | json | split(level) on "-" as nested.nested.boo
+"""
+input = """
+{"level": "INFO", "nested": { "nested": "invalid" } }
+{"level": "INFO", "nested": { "nested": 10 } }
+{"level": "INFO", "nested": { "nested": {} } }
+{"level": "INFO", "nested": { "nested": [] } }
+{"level": "INFO", "nested": { "nested": {"boo": "foo"} } }
+{"level": "INFO", "nested": { "nested": {"abc": 10} } }
+"""
+output = """
+[level=INFO]         [nested={nested:{boo:[INFO]}}]
+[level=INFO]         [nested={nested:{boo:[INFO]}}]
+[level=INFO]         [nested={nested:{abc:10, boo:[INFO]}}]
+"""
+error = """
+error: Expected object, found invalid
```

Comment:
I wonder if this should work like `mkdir -p` and create those keys even if they don't exist. But lets keep the scope under control for now to get this merged